### DISCUSSION TITLE
Help needed on controlling a  Xiomi Aqara ZNLDP12LM smart bulb with MQTT messages.

### DISCUSSION THREAD

**Discussion by nmaou** [arendst/Tasmota#12066] [Q&A] [answered]
Hello,
I own a Xiomi Aqara ZNLDP12LM zigbee smart bulb, which I want to control using the SONOFF zbbgridge zigbee coordinator which I flushed with Tasmota. 
I can control the buld using the following Tasmota commands and the Tasmota console. 
ZbSend { "device":"0x4f11", "send":{"Power":"ON"} }
ZbSend { "device":"0x4f11", "send":{"Dimmer":"254"} } 
ZbSend { "device":"0x4f11", "send":{"CT":"153"} }

I want to control the bulb by sending the commands using MQTT so I can then control the bulb using OpenHab and the MQTT Broker.

As a first step I want to use MQTT Explorer to send the proper MQTT message to Tasmota so as to control the bulb. I read the Tasmota documentation  but I do not understand how to do that.
I do not know:
(1) what topic to use
I use NMAuto as the value of %topic% in the MQTT confoguration section so I assume the topic should start with “cmnd/NMAuto” but it do not know how to create a topic that points to the bulb.
(2)what message to send. 
The fulltopic is set to “%prefix%/%topic%/”

Any comments that can point me in the right direction are welcomed. 

Napoleon Maou

**Comment by sfromis** [comment, marked as answer]
Unless you changed your setup defaults, the topic would be `cmnd/NMAuto/Backlog` and then the full command as payload. You can also use topic `cmnd/NMAuto/ZbSend` and the command arguments as payload. You can verify your topic by activating `Weblog 3` and restarting while in the console. At boot time, Tasmota will produce messages about the MQTT subscriptions.

The individual bulb is addressed using the "device" part of the JSON payload you build. Let me recommend you to use `ZbName` to get a readable name, and just not at all use that short address. In addition to much improved readability, the short address could also change if you for some reason would have to reconfigure the Zigbee setup from start.

**Reply by nmaou** [reply]
Thanks for the prompt reply.
I tried using the topic cmnd/NMAuto/ZbSend and it works fine.
I also used Weblogic 3 and then restarted the zbbridge to record the message in the console. As you can see in the messages the only message related to the smart bulb are the once in the tele/NMAuto/Sensor topic send when I turn on/off the bulb.
17:51:01.479 **MQT: tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Reachable":false}}}**
17:52:02.218 **MQT: tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Power":0,"0006/F000":117440516,"Endpoint":1,"LinkQuality":65}}}**
17:52:02.616 MQT: **tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Power":1,"Endpoint":1,"LinkQuality":65}}}**
17:52:03.702 MQT: **tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Power":0,"0006/F000":5181696,"Dimmer":254,"0008/F000":5181696,"Endpoint":1,"LinkQuality":65}}}**


And I wonder if the  cmnd/NMAuto/ZbSend topic is the way to control all bulbs and switches I will put on the zigbee network.

**Napoleon Maou**


00:00:00.001 HDW: ESP8266EX
00:00:00.035 CFG: Loaded from flash at 1FB, Count 96
00:00:00.038 CFG: CR 352/699, Busy 0
00:00:00.040 ZIG: ZBBridge EEPROM found at address 0x50
00:00:00.041 SNS: Hardware Serial
00:00:00.042 SRC: Restart
00:00:00.043 Project tasmota NMZbBridge Version 9.4.0(zbbridge)-2_7_4_9(2021-04-23T10:07:24)
00:00:00.044 ZIG: Resetting EZSP device
00:00:00.193 ZbEZSPReceived: time-out, discarding F8
00:00:00.197 WIF: Attempting connection...
00:00:00.533 WIF: Connecting to AP1 Nova-66AEC3 Channel 11 BSSId A4:91:B1:66:AE:C3 in mode 11n as NMAuto-1426...
00:00:01.752 WIF: Attempting connection...
00:00:02.752 WIF: Attempting connection...
00:00:03.753 WIF: Attempting connection...
00:00:04.753 WIF: Connected
00:00:05.003 HTP: Web server active on NMAuto-1426 with IP address 192.168.1.170
00:00:05.544 RTC: UTC 2021-05-11T16:50:45, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
17:50:46.024 MQT: Attempting connection...
17:50:46.038 MQT: Connected
17:50:46.040 MQT: tele/NMAuto/LWT = Online (retained)
17:50:46.042 MQT: cmnd/NMAuto/POWER = 
17:50:46.043 MQT: Subscribe to cmnd/NMAuto/#
17:50:46.045 MQT: Subscribe to cmnd/tasmotas/#
17:50:46.046 MQT: Subscribe to cmnd/DVES_E2A592_fb/#
17:50:46.050 MQT: tele/NMAuto/INFO1 = {"Info1":{"Module":"Sonoff ZbBridge","Version":"9.4.0(zbbridge)","FallbackTopic":"cmnd/DVES_E2A592_fb/","GroupTopic":"cmnd/tasmotas/"}}
17:50:46.056 MQT: tele/NMAuto/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"NMAuto-1426","IPAddress":"192.168.1.170"}}
17:50:46.061 MQT: tele/NMAuto/INFO3 = {"Info3":{"RestartReason":"Software/System restart"}}
17:50:47.479 HTP: Main Menu
17:50:48.437 APP: Boot Count 41
17:50:48.587 CFG: Saved to flash at FD, Count 97, Bytes 4096
17:50:50.446 MQT: tele/NMAuto/STATE = {"Time":"2021-05-11T17:50:50","Uptime":"0T00:00:12","UptimeSec":12,"Vcc":3.538,"Heap":30,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Nova-66AEC3","BSSId":"A4:91:B1:66:AE:C3","Channel":11,"RSSI":74,"Signal":-63,"LinkCount":1,"Downtime":"0T00:00:06"}}
17:50:55.011 ZIG: Resetting EZSP device
17:50:56.286 MQT: tele/NMAuto/RESULT = {"ZbState":{"Status":1,"Message":"EFR32 booted","RestartReason":"Power-on","Code":2}}
17:50:56.291 ZIG: ZbEZSPSend 000008
17:50:56.335 ZIG: {"ZbEZSPReceived":"000008029067"}
17:50:56.338 MQT: tele/NMAuto/RESULT = {"ZbState":{"Status":55,"Version":"6.7.9.0","Protocol":8,"Stack":2}}
17:50:56.345 MQT: tele/NMAuto/RESULT = {"ZbState":{"Status":3,"Message":"Configured, starting coordinator"}}
17:50:56.350 ZIG: ZbEZSPSend 53001E0200
17:50:56.406 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.407 ZIG: ZbEZSPSend 5300061000
17:50:56.477 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.478 ZIG: ZbEZSPSend 53000C0200
17:50:56.534 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.535 ZIG: ZbEZSPSend 53000D0500
17:50:56.586 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.587 ZIG: ZbEZSPSend 5300112000
17:50:56.659 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.660 ZIG: ZbEZSPSend 530012001E
17:50:56.712 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.713 ZIG: ZbEZSPSend 5300190200
17:50:56.783 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.784 ZIG: ZbEZSPSend 53001A1000
17:50:56.854 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.856 ZIG: ZbEZSPSend 53001E0400
17:50:56.923 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.924 ZIG: ZbEZSPSend 5300220200
17:50:56.978 ZIG: {"ZbEZSPReceived":"530000"}
17:50:56.979 ZIG: ZbEZSPSend 53002A0300
17:50:57.031 ZIG: {"ZbEZSPReceived":"530000"}
17:50:57.032 ZIG: ZbEZSPSend 53002D0100
17:50:57.102 ZIG: {"ZbEZSPReceived":"530000"}
17:50:57.104 ZIG: ZbEZSPSend 530001FF00
17:50:57.175 ZIG: {"ZbEZSPReceived":"000000"}
17:50:57.177 ZIG: ZbEZSPSend 02000104010500000F0F0000040005000600070008000A00020100030004020403040504060400050000040005000600070008000A0002010003000402040304050406040005
17:50:57.233 ZIG: {"ZbEZSPReceived":"020000"}
17:50:57.235 ZIG: ZbEZSPSend 02000B04010500000000
17:50:57.288 ZIG: {"ZbEZSPReceived":"020000"}
17:50:57.289 ZIG: ZbEZSPSend 100000F9FF58020807020500
17:50:57.340 ZIG: {"ZbEZSPReceived":"100000"}
17:50:57.342 ZIG: ZbEZSPSend 55000003
17:50:57.395 ZIG: {"ZbEZSPReceived":"550000"}
17:50:57.396 ZIG: ZbEZSPSend 55000220
17:50:57.447 ZIG: {"ZbEZSPReceived":"550000"}
17:50:57.448 ZIG: ZbEZSPSend 55000330
17:50:57.501 ZIG: {"ZbEZSPReceived":"550000"}
17:50:57.502 ZIG: ZbEZSPSend 55000551
17:50:57.556 ZIG: {"ZbEZSPReceived":"550000"}
17:50:57.557 ZIG: ZbEZSPSend 55000660
17:50:57.608 ZIG: {"ZbEZSPReceived":"550000"}
17:50:57.610 ZIG: ZbEZSPSend 17000000
17:50:57.663 ZIG: {"ZbEZSPReceived":"170000"}
17:50:57.665 ZIG: {"ZbEZSPReceived":"190090"}
17:50:57.667 ZIG: ZbEZSPSend 6A0003
17:50:57.721 ZIG: {"ZbEZSPReceived":"6A0000030003A14097A598F4ABE2A14097A598F4ABE20100020000000000000000000000000000"}
17:50:57.722 ZIG: ZbEZSPSend 2600
17:50:57.775 ZIG: {"ZbEZSPReceived":"2600070409FEFF23A460"}
17:50:57.776 ZIG: ZbEZSPSend 2800
17:50:57.830 ZIG: {"ZbEZSPReceived":"2800000198F4ABE2CCCCCCCC9834140B0000000000F8FF07"}
17:50:57.834 MQT: tele/NMAuto/RESULT = {"ZbState":{"Status":56,"IEEEAddr":"0x60A423FFFE090407","ShortAddr":"0x0000","DeviceType":1}}
17:50:57.840 ZIG: ZbEZSPSend 2600
17:50:57.884 ZIG: {"ZbEZSPReceived":"2600070409FEFF23A460"}
17:50:57.885 ZIG: ZbEZSPSend 2700
17:50:57.937 ZIG: {"ZbEZSPReceived":"27000000"}
17:50:57.939 ZIG: Subscribe to group 0 'ZbListen0 0'
17:50:57.940 ZIG: ZbEZSPSend 64000000000100
17:50:57.992 ZIG: {"ZbEZSPReceived":"640000"}
17:50:57.995 MQT: tele/NMAuto/RESULT = {"ZbState":{"Status":0,"Message":"Started"}}
17:50:57.000 ZIG: Zigbee started
17:50:57.027 ZIG: EEPROM signature 0x6D736154 is correct
17:50:57.032 ZIG: Zigbee device information found in EEPROM (7 devices - 236 bytes)
17:50:57.083 ZIG: Zigbee device data in EEPROM (136 bytes)
17:50:59.159 ZigbeeZCLSend device: 0x4F11, group: 0x0000, endpoint:1, cluster:0x0006, cmd:0x00, send:"0000"
17:50:59.161 ZIG: ZbEZSPSend 340000114F040106000101400100000101050001000000
17:50:59.214 ZIG: {"ZbEZSPReceived":"340000E3"}
17:50:59.263 ZigbeeZCLSend device: 0x4F11, group: 0x0000, endpoint:1, cluster:0x0008, cmd:0x00, send:"0000"
17:50:59.265 ZIG: ZbEZSPSend 340000114F040108000101400100000201050002000000
17:50:59.318 ZIG: {"ZbEZSPReceived":"340000E4"}
17:50:59.369 ZigbeeZCLSend device: 0x4F11, group: 0x0000, endpoint:1, cluster:0x0300, cmd:0x00, send:"000001000300040007000800"
17:50:59.371 ZIG: ZbEZSPSend 340000114F0401000301014001000003010F000300000001000300040007000800
17:50:59.424 ZIG: {"ZbEZSPReceived":"340000E5"}
17:51:01.479 **MQT: tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Reachable":false}}}**
17:51:03.998 ZIG: {"ZbEZSPReceived":"3F0000114F04010600010140000000E3016600"}
17:51:04.101 ZIG: {"ZbEZSPReceived":"3F0000114F04010800010140000000E4016600"}
17:51:04.207 ZIG: {"ZbEZSPReceived":"3F0000114F04010003010140000000E5016600"}
17:51:07.631 ZIG: {"ZbEZSPReceived":"45000500001F0000000000000000FF000000FFFF2A8105A5524322004B1200661B4322004B1200C0AE3521004B120044B36722004B120048562622004B1200"}
17:51:07.639 MQT: tele/NMAuto/RESULT = {"ZbParent":{"Device":"0x0000","Children":5,"ChildInfo":["0x00124B00224352A5","0x00124B0022431B66","0x00124B002135AEC0","0x00124B002267B344","0x00124B0022265648"]}}
17:51:08.596 ZIG: {"ZbEZSPReceived":"3F0006FCFF00001F0000000000000000010000"}
17:52:01.891 SRC: MQTT
17:52:01.893 CMD: Group 0, Index 1, Command "ZBSEND", Data "{ "device":"Door_Light_01", "send":{"Power":"OFF"} }"
17:52:01.896 ZIG: guessing endpoint 1
17:52:01.898 ZigbeeZCLSend device: 0x4F11, group: 0x0000, endpoint:1, cluster:0x0006, cmd:0x00, send:""
17:52:01.900 ZIG: ZbEZSPSend 340000114F04010600010140010000040103110400
17:52:01.904 MQT: stat/NMAuto/RESULT = {"ZbSend":"Done"}
17:52:01.948 ZIG: {"ZbEZSPReceived":"340000E6"}
17:52:01.966 ZIG: {"ZbEZSPReceived":"3F0000114F04010600010140010000E6010000"}
17:52:01.973 ZIG: {"ZbEZSPReceived":"450000040106000101000100003998C2114FFFFF0E18080A0000100000F02304000007"}
17:52:01.976 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0006","srcaddr":"0x4F11","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":65,"securityuse":0,"seqnumber":57,"fc":"0x18","frametype":0,"direction":1,"disableresp":1,"manuf":"0x0000","transact":8,"cmdid":"0x0A","payload":"0000100000F02304000007"}}
17:52:01.979 ZIG: ZbZCLRawReceived: {"0x4F11":{"0006/0000":0,"0006/F000":117440516,"Endpoint":1,"LinkQuality":65}}
17:52:02.124 ZigbeeZCLSend device: 0x4F11, group: 0x0000, endpoint:1, cluster:0x0006, cmd:0x00, send:"0000"
17:52:02.126 ZIG: ZbEZSPSend 340000114F040106000101400100000501050005000000
17:52:02.182 ZIG: {"ZbEZSPReceived":"340000E7"}
17:52:02.197 ZIG: {"ZbEZSPReceived":"3F0000114F04010600010140010000E7010000"}
17:52:02.208 ZIG: {"ZbEZSPReceived":"450000040106000101000100003A98C2114FFFFF081805010000001001"}
17:52:02.210 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0006","srcaddr":"0x4F11","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":65,"securityuse":0,"seqnumber":58,"fc":"0x18","frametype":0,"direction":1,"disableresp":1,"manuf":"0x0000","transact":5,"cmdid":"0x01","payload":"0000001001"}}
17:52:02.213 ZIG: ZbZCLRawReceived: {"0x4F11":{"0006/0000":1,"Endpoint":1,"LinkQuality":65}}
17:52:02.218 **MQT: tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Power":0,"0006/F000":117440516,"Endpoint":1,"LinkQuality":65}}}**
17:52:02.616 MQT: **tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Power":1,"Endpoint":1,"LinkQuality":65}}}**
17:52:03.289 ZIG: {"ZbEZSPReceived":"450000040106000101000100003B98C2114FFFFF0E18090A0000100000F02300114F00"}
17:52:03.291 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0006","srcaddr":"0x4F11","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":65,"securityuse":0,"seqnumber":59,"fc":"0x18","frametype":0,"direction":1,"disableresp":1,"manuf":"0x0000","transact":9,"cmdid":"0x0A","payload":"0000100000F02300114F00"}}
17:52:03.293 ZIG: ZbZCLRawReceived: {"0x4F11":{"0006/0000":0,"0006/F000":5181696,"Endpoint":1,"LinkQuality":65}}
17:52:03.299 ZIG: {"ZbEZSPReceived":"450000040108000101000100003C98C2114FFFFF0E180A0A000020FE00F02300114F00"}
17:52:03.301 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0008","srcaddr":"0x4F11","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":65,"securityuse":0,"seqnumber":60,"fc":"0x18","frametype":0,"direction":1,"disableresp":1,"manuf":"0x0000","transact":10,"cmdid":"0x0A","payload":"000020FE00F02300114F00"}}
17:52:03.303 ZIG: ZbZCLRawReceived: {"0x4F11":{"0008/0000":254,"0008/F000":5181696,"Endpoint":1,"LinkQuality":65}}
17:52:03.702 MQT: **tele/NMAuto/SENSOR = {"ZbReceived":{"0x4F11":{"Device":"0x4F11","Name":"Door_Light_01","Power":0,"0006/F000":5181696,"Dimmer":254,"0008/F000":5181696,"Endpoint":1,"LinkQuality":65}}}**

**Reply by sfromis** [reply]
You can also have a topic structure per device instead of one topic for all devices. Do read the [Zigbee section in commands](https://tasmota.github.io/docs/Commands/#zigbee) carefully, you can improve upon the default options which are really not the best way of running. And let me again recommend, as best practice, to use ZbName for each device instead depending on those short addresses.
